### PR TITLE
[Cassandra]  Fix dashboard issues.

### LINKED ISSUE #3031: [cassandra] [Logs Cassandra] System Logs dashboard is broken
Loading "[Logs Cassandra] System Logs" fails on 8.x (tested 8.2.0-SNAPSHOT). Details on the problem can be found in #3023.

> Error encountered while loading saved dashboard: Could not find reference for kibanaSavedObjectMeta.searchSourceJSON.filter[0].meta.index

Screen Shot 2022-04-07 at 12 06 16

### PR DESCRIPTION
**Pull request by kush-elastic**
- Bugfix

## What does this PR do?

- Fix Cassandra dashboard issues.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/cassandra directory.
- Run the following command to run tests.
`elastic-package test`

## Related issues

- Fixes #3031

### CODE REVIEW

**Comment by yug-elastic** [comment]
@lalit-satapathy Kindly help us add the right reviewer here.
CC: @akshay-saraswat

File: packages/cassandra/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.2.3"
```
Comment (ishleenk17) [inline packages/cassandra/changelog.yml:1]: If we are tagging this as enhancement, we should bump the minor version to 1.3.0
Else we should make it a bugfix.

**Comment by ishleenk17** [comment]
In packages/cassandra/kibana/dashboard/cassandra-25b7d6d0-1c71-11ec-84f1-e1733c643874.json file I still see occurrences of the snapshot version : "version": "7.15.0-SNAPSHOT"

**Comment by kush-elastic** [comment]
> In packages/cassandra/kibana/dashboard/cassandra-25b7d6d0-1c71-11ec-84f1-e1733c643874.json file I still see occurrences of the snapshot version : "version": "7.15.0-SNAPSHOT"

This is not relevant to this PR. we don't need to change it as bug was specific to visualization.

**Comment by ishleenk17** [comment]
> > In packages/cassandra/kibana/dashboard/cassandra-25b7d6d0-1c71-11ec-84f1-e1733c643874.json file I still see occurrences of the snapshot version : "version": "7.15.0-SNAPSHOT"
> 
> This is not relevant to this PR. we don't need to change it as bug was specific to visualisation.

Ideally, we should not have any dashboards with the snapshot version of stack. 
If this is not relevant to this issue, why is it being picked in this PR ?

**Comment by kush-elastic** [comment]
> Ideally, we should not have any dashboards with the snapshot version of stack. If this is not relevant to this issue, why is it being picked in this PR ?

There are actually multiple reasons to crashing of dashboards and main reason was because of [multiple references of index patterns in dashboard's json](https://github.com/elastic/integrations/pull/3287/files#diff-e61910e19ba43000793afc503c10915a2754fe63521f9668fbc189e57e342d46L185-L198). i guess you can refer this [issue](https://github.com/elastic/integrations/issues/3023) and [conversion](https://github.com/elastic/integrations/issues/3023#issuecomment-1091493456).

**Comment by ishleenk17** [comment]
> > Ideally, we should not have any dashboards with the snapshot version of stack. If this is not relevant to this issue, why is it being picked in this PR ?
> 
> There are actually multiple reasons to crashing of dashboards and main reason was because of [multiple references of index patterns in dashboard's json](https://github.com/elastic/integrations/pull/3287/files#diff-e61910e19ba43000793afc503c10915a2754fe63521f9668fbc189e57e342d46L185-L198). i guess you can refer this [issue](https://github.com/elastic/integrations/issues/3023) and [conversation](https://github.com/elastic/integrations/issues/3023#issuecomment-1091493456).

I am not clear with this dashboard change being present in this change. Let's discuss it in today's meeting.

**Comment by kush-elastic** [comment]
> cassandra-25b7d6d0-1c71-11ec-84f1-e1733c643874.json

Sure, we can dog deeper into it in today's sync. FYI, there are a lot of other integrations whose visualizations contain the snapshot version of the stack. One such example: https://github.com/elastic/integrations/blob/5176089e30cf2932d6e5ca7c90caa2ab9a237bee/packages/elastic_agent/kibana/dashboard/elastic_agent-f47f18cc-9c7d-4278-b2ea-a6dee816d395.json#L75

**Comment by ishleenk17** [comment]
> > cassandra-25b7d6d0-1c71-11ec-84f1-e1733c643874.json
> 
> Sure, we can dig deeper into it in today's sync. FYI, there are a lot of other integrations whose visualizations contain the snapshot version of the stack. One such example:
> 
> https://github.com/elastic/integrations/blob/5176089e30cf2932d6e5ca7c90caa2ab9a237bee/packages/elastic_agent/kibana/dashboard/elastic_agent-f47f18cc-9c7d-4278-b2ea-a6dee816d395.json#L75

As discussed yesterday, `please ensure that another issue is opened up where you take care of not using snapshot versions for the dashboards. `This can cause issues (similar to this dashboard issue)

For now approving this.

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**